packages/subgraph-service/contracts/libraries/IndexingAgreement.sol

```diff
@@ -0,0 +1,519 @@
+            allocation.subgraphDeploymentId,
+            params.currentEpoch,
+            tokensCollected,
+            entities,
+            poi,
+            poiEpoch
+        );
+
+        return (wrapper.collectorAgreement.serviceProvider, tokensCollected);
+    }
+
+    function get(Manager storage self, bytes16 agreementId) external view returns (AgreementWrapper memory) {
+        AgreementWrapper memory wrapper = _get(self, agreementId);
+        require(wrapper.collectorAgreement.dataService == address(this), IndexingAgreementNotActive(agreementId));
+
+        return wrapper;
+    }
+
+    function _getManager() internal pure returns (Manager storage $) {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            $.slot := INDEXING_AGREEMENT_MANAGER_STORAGE_LOCATION
+        }
+    }
+
+    function _setTermsV1(Manager storage _manager, bytes16 _agreementId, bytes memory _data) private {
+        IndexingAgreementTermsV1 memory newTerms = Decoder.decodeIndexingAgreementTermsV1(_data);
+        _manager.termsV1[_agreementId].tokensPerSecond = newTerms.tokensPerSecond;
+        _manager.termsV1[_agreementId].tokensPerEntityPerSecond = newTerms.tokensPerEntityPerSecond;
+    }
+
+    function _cancel(
+        Manager storage _manager,
+        bytes16 _agreementId,
+        State memory _agreement,
+        IRecurringCollector.AgreementData memory _collectorAgreement,
+        IRecurringCollector.CancelAgreementBy _cancelBy
+    ) private {
+        delete _manager.allocationToActiveAgreementId[_agreement.allocationId];
+
+        emit IndexingAgreementCanceled(
+            _collectorAgreement.serviceProvider,
+            _collectorAgreement.payer,
+            _agreementId,
+            _cancelBy == IRecurringCollector.CancelAgreementBy.Payer
+                ? _collectorAgreement.payer
+                : _collectorAgreement.serviceProvider
+        );
+
+        _directory().recurringCollector().cancel(_agreementId, _cancelBy);
+    }
+
+    function _tokensToCollect(
+        Manager storage _manager,
+        bytes16 _agreementId,
+        IRecurringCollector.AgreementData memory _agreement,
+        uint256 _entities
+    ) private view returns (uint256) {
+        IndexingAgreementTermsV1 memory termsV1 = _manager.termsV1[_agreementId];
+
+        uint256 collectionSeconds = block.timestamp;
+        collectionSeconds -= _agreement.lastCollectionAt > 0 ? _agreement.lastCollectionAt : _agreement.acceptedAt;
+
+        return collectionSeconds * (termsV1.tokensPerSecond + termsV1.tokensPerEntityPerSecond * _entities);
+    }
+
+    function _isActive(AgreementWrapper memory wrapper) private view returns (bool) {
+        return
+            wrapper.collectorAgreement.dataService == address(this) &&
+            wrapper.collectorAgreement.state == IRecurringCollector.AgreementState.Accepted &&
+            wrapper.agreement.allocationId != address(0);
+    }
```

Comment:
Took me a bit to figure out why we need all this conditions, worth documenting this. Namely that for an agreement to be considered valid it needs to:
- be registered and accepted in the collector for this specific data service
- be registered in this contract